docs/developer-guide/implement-a-udf.rst

```diff
@@ -0,0 +1,349 @@
+.. _implement-a-udf:
+
+Implement a User-defined Function (UDF)
+#######################################
+
+Prerequisites
+     - `Apache Maven <https://maven.apache.org/download.cgi>`__
+     - |cp| installed locally
+     - Internet connectivity for downloading Confluent POM files
+
+Create the KSQL Extensions Directory
+************************************
+
+When you create a custom user-defined function (UDF), you implement it in Java
+and deploy it as a JAR to the KSQL extensions directory. By default, this 
+directory doesn't exist, so you need to create it and assign it in the KSQL
+Server configuration properties.
+
+Create the KSQL extensions directory, ``<path-to-confluent>/etc/ksql/ext``:
+
+.. codewithvars:: bash
+
+    mkdir confluent-|release|/etc/ksql/ext
+
+Edit the ``ksql-server.properties`` configuration file in
+``<path-to-confluent>/etc/ksql`` to add the fully qualified path to the
+``ext`` directory:
+
+.. codewithvars:: text
+
+    ksql.extension.dir=/home/my-home-dir/confluent-|release|/etc/ksql/ext
+
+.. note::
+
+    Use the fully qualified path. KSQL Server won't load extensions if the 
+    path begins with ``~``.
+
+
+Create the Source and Project Files
+***********************************
+
+The following steps shows how to implement your UDF in a Java class and build
+it by defining a Maven POM file.
+
+#. Create a root directory for your UDF's source code and project files.
+#. Create the source code directory, which has a path that corresponds with
+   the package name.
+#. Create the Java source code file in the source code directory. 
+#. Create a Project Object Model (POM) file that defines how Maven builds the
+   source code.
+
+Create the Source Code Directory
+================================
+
+From the root directory for your UDF, create the source code directory. In this
+example, the package name is ``io.confluent.ksql.udfdemo``.

```

Comment:
I would suggest using something that doesn't start with `io.confluent`, as people tend to cut and paste. Really, they should be using their own namespace, not ours.   Maybe use `my.company.ksql.udf.demo`?